Automattic/jetpack
# Mobile stories block

#### Changes proposed in this Pull Request:

This PR adds `.native.*` files to support the Story block rendering on the [gutenberg-mobile](https://github.com/wordpress-mobile/gutenberg-mobile) editor used by the [WordPress Android](https://github.com/wordpress-mobile/WordPress-Android) and [iOS apps](https://github.com/wordpress-mobile/WordPress-iOS). 

Related Gutenberg PR: https://github.com/WordPress/gutenberg/pull/26721
Related Gutenberg-mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2611
Related WPAndroid PR: https://github.com/wordpress-mobile/WordPress-Android/pull/12939 and https://github.com/wordpress-mobile/WordPress-Android/pull/13130
Related WPiOS PR: https://github.com/WordPress/gutenberg/pull/26704

**NOTE**: the code has already been reviewed and merged in several other PRs feeding this feature branch. Omiting climate issues in this PR for now given they're not really relevant.

#### Jetpack product discussion


See pbD5rk-m7 and other links from there.


### Test instructions:

@brbrr  had a few interesting questions to complete the description of the PR that I'll try to address here:

> - Do I need android device to test it?

You can also use an emulator. This will come with Android Studio (see below for instructions to download and install Android Studio)

> - Can I use my iPhone instead?

Technically yes, but for the goals of this set of PRs the short answer is no.

> - Do I need to build app manually, or should I use beta/production version?

We have CI build APKs for testing if you'd prefer to skip the native steps. You can download the CI-built APK corresponding to the [WPAndroid PR here](https://81069-9306568-gh.circle-artifacts.com/0/Artifacts/WordPress-pr-13130-build-81069.apk). 

> - Any way I can test it without mobile device?

Yes, with an emulator. Do all of the following. YMMV but this is basically what you'll need to get it running.


#### Setup (SKIP to "Actual testing..." below if you don't intend to do any sort of debugging)

1. Clone the [WordPress for Android](https://github.com/wordpress-mobile/WordPress-Android) repository locally. Use the branch at this PR https://github.com/wordpress-mobile/WordPress-Android/pull/13130. For cloning please bear in mind you'll need to also clone the submodules so, use this:
`git clone --recurse-submodules git@github.com:wordpress-mobile/WordPress-Android.git`
2. Download and Install[ Android Studio](https://developer.android.com/studio) 
3. Open it and create an android virtual device you'll use as an emulator. Read [managing avds](https://developer.android.com/studio/run/managing-avds ) for instructions.


#### React native part
1. open a terminal in the WordPress-Android folder you just cloned. Do `cd libs/gutenberg-mobile`
2. run  `nvm install --latest-npm`.
3. npm install
4. once it's finished, if you intend only to smoke test that's enough, but if you want to make changes to code you can run the metro server with `npm run start:reset`. You will have to make a change in the WAndroid  project's `gradle.properties` file as well.

#### WordPress for Android part
1. if you want to be able to edit gutenberg / jetpack code for reasons, include the following line in your `gradle.properties` file:
```
wp.BUILD_GUTENBERG_FROM_SOURCE = true
```
2. To build the project, follow the build instructions at https://github.com/wordpress-mobile/WordPress-Android#wordpress-for-android (or just open it on Android Studio and hit the Run button)
3. Once you have a running build, you should be able to follow the steps provided in https://github.com/wordpress-mobile/WordPress-Android/pull/12939 for smoke testing the functionality, but a shorter version of the steps to try the functionality are given below as well.

#### Actual Stories functionality testing:

a. tap on the main screen's "+" floating action button, then choose `Story`.
b. Take a picture or select from the media picker to create some story slides.
c. Then, tap on the "->" icon on the top-right of the screen in the visual story composer, and give your Story a name, then hit the big PUBLISH button at the bottom.
d. Wait for it to upload.
e. Now go to your Posts list, and look for the post you just created
f. open it by tapping on it
g. the Gutenberg editor should load and the Story block should render.
h. tap on it and your story should be loaded in the Story creator again for editing.


Here's a GIF file showing all of these steps and a few more.

![testStory](https://user-images.githubusercontent.com/6597771/98406479-cb93da80-204c-11eb-858e-9704b14c029c.gif)



#### Proposed changelog entry for your changes:

no changes visible to the web side of things, this PR only affects the WordPress mobile apps
